tests/testLsstOpticalDistortions.py

```diff
@@ -0,0 +1,1072 @@
+
+        rng = np.random.RandomState(1845332)
+
+        for band in 'ugrizy':
+
+            x_pix, y_pix = pixelCoordsFromRaDecLSST(self._truth_data['ra'],
+                                                    self._truth_data['dec'],
+                                                    pm_ra=self._truth_data['pmra'],
+                                                    pm_dec=self._truth_data['pmdec'],
+                                                    parallax=self._truth_data['px'],
+                                                    v_rad=self._truth_data['vrad'],
+                                                    obs_metadata=self._obs,
+                                                    band=band)
+
+            self.assertIsInstance(x_pix, np.ndarray)
+            self.assertIsInstance(y_pix, np.ndarray)
+
+            n_nan = 0
+            n_good = 0
+            subsample = rng.choice(np.arange(len(x_pix), dtype=int),
+                                   size=50, replace=False)
+            for ii in subsample:  # only test on a subsample; this is slow
+                x_pix1, y_pix1 = pixelCoordsFromRaDecLSST(self._truth_data['ra'][ii],
+                                                          self._truth_data['dec'][ii],
+                                                          pm_ra=self._truth_data['pmra'][ii],
+                                                          pm_dec=self._truth_data['pmdec'][ii],
+                                                          parallax=self._truth_data['px'][ii],
+                                                          v_rad=self._truth_data['vrad'][ii],
+                                                          obs_metadata=self._obs,
+                                                          band=band)
+
+                self.assertIsInstance(x_pix1, numbers.Number)
+                self.assertIsInstance(y_pix1, numbers.Number)
+                if not np.isnan(x_pix1):
+                    self.assertAlmostEqual(x_pix1, x_pix[ii], 8)
+                    self.assertAlmostEqual(y_pix1, y_pix[ii], 8)
+                    n_good += 1
+                else:
+                    n_nan += 1
+
+            self.assertLess(n_nan, n_good//2)
+
+    def test_pupil_coords_from_focal_plane_coords_LSST(self):
+        """
+        Test that pupilCoordsFromFocalPlaneCoordsLSST inverts
+        focalPlaneCoordsFromPupilCoordsLSST
+        """
+        x_f = np.arange(-400.0, 400.0, 20.0)
+        y_f = np.arange(-400.0, 400.0, 20.0)
+        mesh = np.meshgrid(x_f, y_f)
+        x_f = mesh[0].flatten()
+        y_f = mesh[1].flatten()
+        for band in 'ugrizy':
+            x_p, y_p = pupilCoordsFromFocalPlaneCoordsLSST(x_f, y_f, band=band)
+            x_f1, y_f1 = focalPlaneCoordsFromPupilCoordsLSST(x_p, y_p,
+                                                             band=band)
+
+            dd = np.sqrt((x_f-x_f1)**2 + (y_f-y_f1)**2)
+            self.assertLess(dd.max(), 1.0e-5)
```

Comment:
Do you think we should have the test match closer to the max value mentioned in the documentation of the methods? There you said up to 2.18e-6 mm.

Reply:
I'll bump it down to 3.0e-6;  2.18e-6 feels like too many sigfigs to me